data/transactions/logic/eval.go

```diff
@@ -1985,6 +1985,28 @@ func branchTarget(cx *EvalContext) (int, error) {
 	return target, nil
 }
 
+func branchSwitch(cx *EvalContext, branchIdx uint64) (int, uint64, error) {
+	numOffsets, bytesUsed := binary.Uvarint(cx.program[cx.pc+1:])
+	if numOffsets == 0 {
+		return 0, 0, fmt.Errorf("number of offsets must be greater than 0")
+	}
+	if branchIdx >= numOffsets {
+		return 0, 0, fmt.Errorf("provided branch index %d exceeds max offset index %d", branchIdx, numOffsets-1)
+	}
+
+	pos := uint64(cx.pc+1+bytesUsed) + (2 * branchIdx)
+	offset := int16(uint16(cx.program[pos])<<8 | uint16(cx.program[pos+1]))
```

Comment:
Need to precheck the `pos + 1 <= len(cx.program)` so we can't panic from bad bytecode.